### PR TITLE
fix(windows): a bunch of install test fixes

### PR DESCRIPTION
**Pull request by dylan-conway**
### How did you verify your code works?
ci

### CODE REVIEW

File: src/install/install.zig [hunk 1]
```diff
@@ -1910,6 +1910,7 @@ const Waiter = struct {
 // 2.
 pub const PackageManager = struct {
     cache_directory_: ?std.fs.Dir = null,
+    cache_directory_path: string = "",
```
Comment (Jarred-Sumner) [inline src/install/install.zig:1915]: Please add a comment to remove this once we move away from using std.ChildProccess for `git clone`.

File: test/cli/install/registry/bun-install-registry.test.ts [hunk 2]
```diff
@@ -14,25 +14,25 @@ var testCounter: number = 0;
...
-        done();
-      }
-    });
-  });
-});
-
-afterAll(() => {
-  verdaccioServer.kill();
-});
+// beforeAll(async () => {
```
Comment (Jarred-Sumner) [inline test/cli/install/registry/bun-install-registry.test.ts]: was disabling verdaccio (and subsequently, these tests) intentional?

File: test/harness.ts [hunk 3]
```diff
@@ -249,3 +250,5 @@ expect.extend({
     };
   },
 });
+
+export const pathSep = platform() === "win32" ? "\\" : "/";
```
Comment (Jarred-Sumner) [inline test/harness.ts]:
```suggestion
export const {sep as pathSep} from 'node:path'; 
```

File: test/cli/install/registry/bun-install-registry.test.ts (hunk 2 above)
Comment (dylan-conway) [reply test/cli/install/registry/bun-install-registry.test.ts]: Not intentional, for testing locally

**Pull request merged**